### DISCUSSION TITLE
Add support for MySQL socket connections

### DISCUSSION THREAD

**Discussion by benjamin-smith** [directus/directus#17669] [Feature Requests]
### Summary

Allowing MySQL connections via unix sockets would allow for increased flexibility. This was possible in Directus 8.

As far as I can tell, `DB_HOST` is a required variable for MySQL connections. I do not see any options for connecting via a socket.

### Basic Example

I envision an ENV var:

`DB_SOCKET = "project-id:us-east1:db-instance"`

### Motivation

My use case is I am trying to deploy on Google Cloud Run. Connections to Cloud SQL are recommended via a socket. The MySQL package supports the `socketPath` config option already: https://github.com/mysqljs/mysql#connection-options

### Detailed Design

I don't have much to add here. I suggest the MySQL project's config note be followed for cases when folks specify both Host and Socket ENV vars:

> socketPath: The path to a unix domain socket to connect to. When used host and port are ignored.

### Must Have:
- A new ENV var to allow connections via MySQL socket

### Drawbacks

The only downside AFAICT would be the time it takes to implement.

### Alternatives

For situations where a socket is the only connection method, this would be a blocker for someone adopting Directus. I personally have an alternative (DB_HOST), but, it is inconvenient and will be less secure as I'll have to loosen my firewall.

### Adoption Strategy

Not a breaking change. Just an additional config option / ENV var.

**Comment by benjamin-smith** [comment]
This can be closed. I searched for relevant issues before posting, but not "Discussions".

https://github.com/directus/directus/discussions/12946

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
